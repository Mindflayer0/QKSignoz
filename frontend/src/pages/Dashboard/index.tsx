import getAll from 'api/dashboard/getAll';
import Spinner from 'components/Spinner';
import ListOfAllDashboard from 'container/ListOfDashboard';
import useFetch from 'hooks/useFetch';
import React, { useEffect } from 'react';
import { connect, useSelector } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { GetAllDashboards } from 'store/actions';
import { AppState } from 'store/reducers';
import AppActions from 'types/actions';
import DashboardReducer from 'types/reducer/dashboards';

const Dashboard = ({ getAllDashboards }: DashboardProps): JSX.Element => {
	const { loading, dashboards } = useSelector<AppState, DashboardReducer>(
		(state) => state.dashboards,
	);

	useEffect(() => {
		getAllDashboards();
	}, []);

	if (loading) {
		return <Spinner size="large" tip="Loading.." />;
	}

	return <ListOfAllDashboard listOfDashboards={dashboards} />;
};

interface DispatchProps {
	getAllDashboards: () => void;
}

const mapDispatchToProps = (
	dispatch: ThunkDispatch<unknown, unknown, AppActions>,
): DispatchProps => ({
	getAllDashboards: bindActionCreators(GetAllDashboards, dispatch),
});

type DashboardProps = DispatchProps;

export default connect(null, mapDispatchToProps)(Dashboard);
