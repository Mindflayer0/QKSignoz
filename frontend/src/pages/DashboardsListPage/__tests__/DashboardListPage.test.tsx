import ROUTES from 'constants/routes';
import DashboardsList from 'container/ListOfDashboard';
import { DashboardProvider } from 'providers/Dashboard/Dashboard';
import { MemoryRouter } from 'react-router-dom';
import { render, waitFor } from 'tests/test-utils';

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useLocation: (): { pathname: string } => ({
		pathname: `${process.env.FRONTEND_API_ENDPOINT}/${ROUTES.ALL_DASHBOARD}/`,
	}),
}));

describe('dashboard list page', () => {
	it.skip('should render the list even when the columnKey or the order is mismatched', async () => {
		const { container } = render(
			<MemoryRouter
				initialEntries={['/dashbords?columnKey=asgard&order=stones&page=1']}
			>
				<DashboardProvider>
					<DashboardsList />
				</DashboardProvider>
			</MemoryRouter>,
		);

		await waitFor(() => expect(container).toMatchSnapshot());
	});
});
