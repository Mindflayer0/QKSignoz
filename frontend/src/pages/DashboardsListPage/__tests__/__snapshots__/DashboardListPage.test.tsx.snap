// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard list page should render the list even when the columnKey or the order is mismatched 1`] = `
<div>
   
  <div
    class="dashboards-list-container"
  >
    <div
      class="dashboards-list-view-content"
    >
      <div
        class="dashboards-list-title-container"
      >
        <h1
          class="ant-typography title css-dev-only-do-not-override-2i2tap"
        >
          Dashboards
        </h1>
        <div
          class="ant-flex css-dev-only-do-not-override-2i2tap ant-flex-align-center ant-flex-justify-space-between"
        >
          <span
            class="ant-typography subtitle css-dev-only-do-not-override-2i2tap"
          >
            Create and manage dashboards for your workspace.
          </span>
        </div>
      </div>
      <div
        class="loading-dashboard-details"
      >
        <div
          class="ant-skeleton ant-skeleton-element ant-skeleton-active skeleton-1 css-dev-only-do-not-override-2i2tap"
        >
          <span
            class="ant-skeleton-input ant-skeleton-input-lg skeleton-1"
          />
        </div>
        <div
          class="ant-skeleton ant-skeleton-element ant-skeleton-active skeleton-1 css-dev-only-do-not-override-2i2tap"
        >
          <span
            class="ant-skeleton-input ant-skeleton-input-lg skeleton-1"
          />
        </div>
        <div
          class="ant-skeleton ant-skeleton-element ant-skeleton-active skeleton-1 css-dev-only-do-not-override-2i2tap"
        >
          <span
            class="ant-skeleton-input ant-skeleton-input-lg skeleton-1"
          />
        </div>
        <div
          class="ant-skeleton ant-skeleton-element ant-skeleton-active skeleton-1 css-dev-only-do-not-override-2i2tap"
        >
          <span
            class="ant-skeleton-input ant-skeleton-input-lg skeleton-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
